### PR TITLE
docs: clarify workflow using clusterctl and kubectl 

### PR DESCRIPTION
**Pull request by andrewsykim**
**What this PR does / why we need it**:
* clarifies assumptions and expectations around clusterctl and kubectl
* clean up project README

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: docs/getting_started.md [hunk 1]
```diff
@@ -166,14 +166,26 @@ path that it ran (i.e. `out/kubeconfig`). This is the **admin** kubeconfig file
 going forward to spin up multiple clusters using Cluster API, however, it is recommended that you create dedicated roles
 with limited access before doing so.
 
+Note that from this point forward, you no longer need to use `clusterctl` to provision clusters since your management cluster
+(the cluster used to manage clusters) has been created. Clusters should be provisioned by applying Cluster API resources
```
Comment (mylesagray) [inline docs/getting_started.md]: (the cluster used to manage _workload_ clusters) has been created. Workload clusters...

File: docs/getting_started.md [hunk 2]
```diff
@@ -269,7 +281,17 @@ spec:
...
+$ cd out/prod-workload
+$ kubectl apply -f cluster.yaml
+cluster.cluster.k8s.io/prod-workload created
+$ kubectl apply -f machines.yaml
+machine.cluster.k8s.io/prod-workload-controlplane-1 created
+$ kubectl apply -f machineset.yaml
+machineset.cluster.k8s.io/prod-workload-machineset-1 created
+```
+
 Clusters that are provisioned by the management cluster that run your application workloads are called [Workload Clusters](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/book/GLOSSARY.md#workload-cluster).
```
Comment (mylesagray) [inline docs/getting_started.md]: This should be at the top - a workload vs mgmt distinction needs to be made before talking about each in the doc.

Comment (andrewsykim) [reply docs/getting_started.md]: Yeah this one is tricky, there's a lot of cluster API terms here and I link to the cluster api docs at the top of the doc which has everything defined. Wondering how much of that we need to redefine here

**Pull request merged**